performance/admin.py

```diff
@@ -0,0 +1,404 @@
+                annual_aggregate_output='',
+                annual_pre_audit_output=indicator_data['PrelimaryAudited_Summary2'],
+                annual_deviation_reason=indicator_data['ReasonforDeviation_Summary'],
+                annual_corrective_action=indicator_data['CorrectiveAction_Summary'],
+                annual_otp_comments=indicator_data['OTP_Summary'],
+                annual_national_comments=indicator_data['National_Summary'],
+                annual_dpme_coordincator_comments='',
+                annual_treasury_comments='',
+                annual_audited_output=indicator_data['ValidatedAudited_Summary2'],
+
+                sector=indicator_data['Sector'],
+                programme_name=indicator_data['Programme'],
+                subprogramme_name=indicator_data['SubProgramme'],
+                frequency=indicator_data['Frequency'],
+                type=indicator_data['Type'],
+                subtype=indicator_data['SubType'],
+                mtsf_outcome=indicator_data['Outcome'],
+                cluster=indicator_data['Cluster'],
+                uid=indicator_data['UID'],
+            )
+            num_imported = num_imported + 1
+        elif department_name not in not_matching_departments:
+            not_matching_departments.append(department_name)
+
+    obj_to_update = models.EQPRSFileUpload.objects.get(id=obj_id)
+
+    obj_to_update.num_imported = num_imported
+    obj_to_update.num_not_imported = total_record_count - num_imported
+    obj_to_update.import_report = generate_import_report(True, None, not_matching_departments)
+    obj_to_update.save()
+
+
+def validate_report_type(full_text, obj_id):
+    validated = False
+    for report_type in VALID_REPORT_TYPES:
+        validated = validated or (report_type in full_text)
+
+    if not validated:
+        obj_to_update = models.EQPRSFileUpload.objects.get(id=obj_id)
+        obj_to_update.num_imported = None
+        obj_to_update.num_not_imported = None
+        obj_to_update.import_report = generate_import_report(False, None, [])
+        obj_to_update.save()
+
+    return validated
+
+
+def validate_frictionless(data, obj_id):
+    report = validate(data)
+    validated = report.valid
+    if not validated:
+        obj_to_update = models.EQPRSFileUpload.objects.get(id=obj_id)
+        obj_to_update.num_imported = None
+        obj_to_update.num_not_imported = None
+        obj_to_update.import_report = generate_import_report(True, report, [])
+        obj_to_update.save()
+
+    return validated
+
+
+def parse_and_process_csv(full_text, obj_id):
+    report_type_validated = validate_report_type(full_text, obj_id)
+    if report_type_validated:
+        clean_text = full_text.split('\n', 3)[3]
+        f = StringIO(clean_text)
+        reader = csv.DictReader(f)
+        parsed_data = list(reader)
+
+        task = async_task(func=save_imported_indicators, parsed_data=parsed_data, obj_id=obj_id)
```

Comment:
I don't think we should store the entire file contents in a task argument - then we end up storing all of this in the task database table. Can the async task instead read the file via obj_id?

Reply:
remember we want to push work to the async task and do as little as possible synchronously on save

Reply:
done